packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Models/Types/SystemModelSerializationTypeProvider.cs

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Microsoft.Generator.CSharp.Expressions;
+
+namespace Microsoft.Generator.CSharp.ClientModel
+{
+    /// <summary>
+    /// This class provides the set of serialization models, methods, and interfaces for a given model.
+    /// </summary>
+    internal sealed class SystemModelSerializationTypeProvider : TypeProvider
+    {
+        internal static readonly Parameter SerializationOptionsParameter =
+            new("options", null, typeof(ModelReaderWriterOptions), null, ValidationType.None, null);
+
+        public SystemModelSerializationTypeProvider(ModelTypeProvider model) : base(null)
+        {
+            Model = model;
+            Name = model.Name;
+            Json = BuildJsonSerialization();
+            // Initialize the serialization interfaces
+            IJsonModelTInterface = new CSharpType(typeof(IJsonModel<>), Model.Type);
+            IJsonModelObjectInterface = Model.IsStruct ? new CSharpType(typeof(IJsonModel<object>)) : null;
+            IPersistableModelTInterface = new CSharpType(typeof(IPersistableModel<>), Model.Type);
+            IPersistableModelObjectInterface = Model.IsStruct ? (CSharpType)typeof(IPersistableModel<object>) : null;
+        }
+
+        public override string Name { get; }
+        public ModelTypeProvider Model { get; }
+        public JsonObjectSerialization? Json { get; }
+        public CSharpType IJsonModelTInterface { get; }
```

Comment:
Do we need these properties?

Reply:
Not necessarily. I would like these interface types to be defined within the class so it can be easier to change in fewer areas if we ever need to do so. I will change these to private fields instead since there is no need to have them as properties right now.